### ISSUE TITLE
2.1.0 and 2.1.1 uploadFile failing

### ISSUE THREAD

**Issue by Georgegriff** [puppeteer/puppeteer#5420]
I have this code: (via codeceptjs)
```
    async attachFile(locator, pathToFile) {
      const file = path.join(global.codecept_dir, pathToFile);

      if (!fileExists(file)) {
        throw new Error(`File at ${file} can not be found on local system`);
      }
      const els = await findFields.call(this, locator);
      assertElementExists(els, 'Field');
      await els[0].uploadFile(file);
      return this._waitForAction();
    }
```

In in 2.0.0 this works but in 2.1.0 and 2.1.1 this no longer works.

**Comment by Georgegriff** [comment]
Duplicate: https://github.com/puppeteer/puppeteer/issues/5363

**Comment by gziolo** [comment]
I can confirm that it works properly with `2.0.0` but it doesn't work with `2.1.1`:
https://github.com/WordPress/gutenberg/pull/20268#issuecomment-587023772

**Comment by swftvsn** [comment]
Same here, uploads fail with 2.1.1

**Comment by yurynix** [comment]
This works for me as workaround on:
80.0.3987.0 puppeteer-core version: 2.1.1
79.0.3945.0 puppeteer-core version: 2.1.0
```es6
  const uploadInput = await page.$(someSelector)
  const filePath = path.join(__dirname, fileRelativePath);
  await uploadInput.uploadFile(filePath);
  await page.evaluate((inputSelector) => {
    document.querySelector(inputSelector).dispatchEvent(new Event('change', { bubbles: true }));
  }, someSelector)
```

So maybe something like https://github.com/puppeteer/puppeteer/pull/5446 is in order?

**Comment by DJ-Glock** [comment]
Same issue. And workaround with uploadFile for input element does not work for our site, because we use custom inputs. Anyway it needs to be fixed.

**Comment by sfedorin** [comment]
Hi everyone. Guys, is there any progress?

**Comment by paullewis** [comment]
This should be fixed as of 532ae573d29063ffd081ed9ee340d71b25320dec as we changed the underlying implementation to use `DOM.setFileInputFiles`, which behaves a lot more as expected. We also fire events for `change` and `input`, too. Going to close this issue, but please let us know if it's not fixed.
